src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +457,66 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Info(nameof(ConfigureAutoLogonIfNeeded));
+
+            if(!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if(autoLogonConfigManager.RestartNeeded())
```

Comment:
Same comment as UnconfigAutoLogonIfNeeded. Inline and follow same patterns. Move entire if-restart-needed block behind autoLogonConfigManager.Config()